### ISSUE TITLE
Client proxy generation at compile time

### ISSUE THREAD

**Issue by ForkBug** [microsoft/vs-streamjsonrpc#746]
Client proxy generation may be done at compile time via source generator.
This modification could improve performance and add strong typed request for AOT environment(such as xamarin.ios)

**Comment by AArnott** [comment]
This is true. The perf improvement would only be to startup time -- overall throughput would not be affected. 

If we do this, we'd probably add a version constant to the generated class so that at runtime the library may choose to ignore the AOT generated class when it's too old to be used by the current version of the library, since these proxies do occasionally change.

**Comment by moustakakisCon** [comment]
hello, I am integrating streamjsonrpc in a unity game and I am looking to use strong typed request, but `System.Reflection.Emit.AssemblyBuilder.DefineDynamicAssembly` is not supported in IL2CPP scripting backend.

Is there any chance this enhancement will be implemented in the (not so far) future?

**Comment by AArnott** [comment]
Unlikely. Strict AOT environments like unity are decidedly out of scope for anything likely to be prioritized in StreamJsonRpc for the forseeable future. Thanks for asking, @moustakakisCon.
